example/.gitignore

```diff
@@ -1,41 +1,162 @@
-# Miscellaneous
+
+# Created by https://www.toptal.com/developers/gitignore/api/android,flutter
+# Edit at https://www.toptal.com/developers/gitignore?templates=android,flutter
+
+### Android ###
+# Built application files
+*.apk
+*.aar
+*.ap_
+*.aab
+
+# Files for the ART/Dalvik VM
+*.dex
+
+# Java class files
 *.class
+
+# Generated files
+bin/
+gen/
+out/
+#  Uncomment the following line in case you need and you don't have the release build type files in your app
+# release/
+
+# Gradle files
+.gradle/
+build/
+
+# Local configuration file (sdk path, etc)
+local.properties
+
+# Proguard folder generated by Eclipse
+proguard/
+
+# Log Files
 *.log
-*.pyc
-*.swp
-.DS_Store
-.atom/
-.buildlog/
-.history
-.svn/
-
-# IntelliJ related
+
+# Android Studio Navigation editor temp files
+.navigation/
+
+# Android Studio captures folder
+captures/
+
+# IntelliJ
 *.iml
-*.ipr
-*.iws
-.idea/
+.idea/workspace.xml
+.idea/tasks.xml
+.idea/gradle.xml
+.idea/assetWizardSettings.xml
+.idea/dictionaries
+.idea/libraries
+# Android Studio 3 in .gitignore file.
+.idea/caches
+.idea/modules.xml
+# Comment next line if keeping position of elements in Navigation Editor is relevant for you
+.idea/navEditor.xml
```

Comment:
these all are equivalent as .idea/ ??

Reply:
Please note that I am simply using the [android gitignore template](https://github.com/toptal/gitignore/blob/master/templates/Android.gitignore), and of course the [Flutter gitignore template](https://github.com/toptal/gitignore/blob/master/templates/Flutter.gitignore).  I _yield_ to the _wisdom_ of the template maintainers.

Honestly, I expected to get away with just `flutter`, but then it showed untracked files:
```
	.idea/
	android/example_android.iml
	example.iml
```
As a result, I reluctantly added the `android` template too.

Reply:
Yes, idk why they want to split up stuff. 

This is the .gitignore files for [plugin](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/package/.gitignore.tmpl), [app](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/app/.gitignore.tmpl), [android](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/app/android.tmpl/.gitignore) and [ios](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/app/ios.tmpl/.gitignore)

This is what default you get from flutter create command
What you have done is also fine👍